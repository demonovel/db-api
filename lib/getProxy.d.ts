export declare function getProxy(proxy?: string): string;
export default getProxy;
