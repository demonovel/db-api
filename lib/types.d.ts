/**
 * Created by user on 2020/5/18.
 */
import { RequestInit, Response } from 'node-fetch';
export interface IRawRecordRow<T> {
    error: boolean;
    timestamp: number;
    data: T;
}
export interface IFileRecordRow<T extends IFileRecordData = IFileRecordData> extends IRawRecordRow<T> {
}
export interface IFileRecordData {
    timestamp: number;
    exists: true;
    filename: string;
    href: string;
}
export interface IFetchRecordOptionsBase {
    fetchOptions?: RequestInit;
    server?: string;
    retry?: number;
    tapCheck?(r: Response): any;
    tapError?(e: any): any;
}
export declare enum EnumApiType {
    file = "file",
    raw = "raw"
}
