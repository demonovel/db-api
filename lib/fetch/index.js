"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.fetch = void 0;
const fetch_1 = __importDefault(require("./fetch"));
const bluebird_1 = __importDefault(require("bluebird"));
function fetch(url, opts = {}, rawOptions) {
    var _a, _b;
    let signal = opts === null || opts === void 0 ? void 0 : opts.signal;
    opts = {
        method: 'POST',
        ...opts,
        headers: {
            'Accept': 'application/json',
            'Content-Type': 'application/json',
            ...opts.headers,
        },
        signal,
    };
    const fn = () => {
        opts.signal = signal;
        return fetch_1.default(url, opts);
    };
    rawOptions = {
        ...rawOptions,
        tapCheck: (_a = rawOptions === null || rawOptions === void 0 ? void 0 : rawOptions.tapCheck) !== null && _a !== void 0 ? _a : (() => { }),
        tapError: (_b = rawOptions === null || rawOptions === void 0 ? void 0 : rawOptions.tapError) !== null && _b !== void 0 ? _b : (() => { }),
    };
    return _reDo(fn, rawOptions)
        .then((response) => response.json());
}
exports.fetch = fetch;
function _reDo(fn, rawOptions) {
    var _a;
    let retry = ((_a = rawOptions.retry) !== null && _a !== void 0 ? _a : 1) | 0;
    return bluebird_1.default.resolve()
        .then(fn)
        .tapCatch(rawOptions.tapError)
        .tap(rawOptions.tapCheck)
        .catch(e => {
        if (retry-- > 0) {
            return _reDo(fn, {
                ...rawOptions,
                retry,
            });
        }
        return Promise.reject(e);
    });
}
exports.default = fetch;
//# sourceMappingURL=index.js.map