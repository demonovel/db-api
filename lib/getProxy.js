"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getProxy = void 0;
function getProxy(proxy) {
    const env = typeof process !== 'undefined' ? process.env : {};
    return proxy || env.HTTPS_PROXY || env.HTTP_PROXY || void 0;
}
exports.getProxy = getProxy;
exports.default = getProxy;
//# sourceMappingURL=getProxy.js.map