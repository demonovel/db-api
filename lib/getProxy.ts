export function getProxy(proxy?: string)
{
	const env = typeof process !== 'undefined' ? process.env : {};
	return proxy || env["HTTPS_PROXY"] || env["HTTP_PROXY"] || void 0;
}

export default getProxy
