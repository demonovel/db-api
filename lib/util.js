"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.newLinkURL = exports.newFileURL = void 0;
const tslib_1 = require("tslib");
const hash_sum_1 = tslib_1.__importDefault(require("hash-sum"));
const types_1 = require("./types");
//const server01 = `https://api-file-server.epub.now.sh/`;
const server01 = `https://api-file-server.vercel.app/`;
function newFileURL(rootKey, dataKey, server = server01, type = types_1.EnumApiType.file) {
    return new URL(`db/${type !== null && type !== void 0 ? type : null}/${rootKey}/${(0, hash_sum_1.default)(dataKey)}`, server);
}
exports.newFileURL = newFileURL;
function newLinkURL(rootKey, dataKey, server = server01) {
    return new URL(`file/${rootKey}/${(0, hash_sum_1.default)(dataKey)}`, server);
}
exports.newLinkURL = newLinkURL;
//# sourceMappingURL=util.js.map