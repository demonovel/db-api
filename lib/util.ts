import hashSum from 'hash-sum';
import { EnumApiType } from './types';

//const server01 = `https://api-file-server.epub.now.sh/`;
const server01 = `https://api-file-server.vercel.app/`;

export function newFileURL(rootKey: string, dataKey: string | number, server = server01, type: EnumApiType = EnumApiType.file)
{
	return new URL(`db/${type ?? null}/${rootKey}/${hashSum(dataKey)}`, server);
}

export function newLinkURL(rootKey: string, dataKey: string | number, server = server01)
{
	return new URL(`file/${rootKey}/${hashSum(dataKey)}`, server);
}
