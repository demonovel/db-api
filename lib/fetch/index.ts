import { IFileRecordRow } from '../types';
import { RequestInit } from 'node-fetch';
import { newFileURL } from '../util';
import _fetch from './fetch';
import Bluebird from 'bluebird';

export function fetch<T>(url: string, opts: RequestInit = {}): Bluebird<T>
{
	return _fetch(url, {
		method: 'POST',
		...opts,
		headers: {
			'Accept': 'application/json',
			'Content-Type': 'application/json',
			...opts.headers,
		},
	})
		.then((response) => response.json())
}

export default fetch
