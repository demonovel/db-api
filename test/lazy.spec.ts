import { newFileURL, newLinkURL } from '../lib/util';

describe(`describe`, () =>
{
	const siteID = `wenku8` as const;
	const novelID = 2451 as const;

	test(`url`, () =>
	{

		let u1 = newFileURL(siteID, novelID);
		let u2 = newLinkURL(siteID, novelID);

		expect(u1.toString()).toStrictEqual('https://api-file-server.vercel.app/db/file/wenku8/3007a3f8');
		expect(u2.toString()).toStrictEqual('https://api-file-server.vercel.app/file/wenku8/3007a3f8');

		expect(u1).toMatchSnapshot();
		expect(u2).toMatchSnapshot();

	});

})
