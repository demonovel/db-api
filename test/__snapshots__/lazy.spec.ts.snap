// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describe url 1`] = `"https://api-file-server.vercel.app/db/file/wenku8/3007a3f8"`;

exports[`describe url 2`] = `"https://api-file-server.vercel.app/file/wenku8/3007a3f8"`;
